backend/src/main/java/com/bakdata/conquery/models/preproc/Preprocessed.java

```diff
@@ -50,26 +50,69 @@ public Preprocessed(PreprocessingConfig config, ImportDescriptor descriptor) thr
 		if(!(primaryColumn.getParser() instanceof StringParser)) {
 			throw new IllegalStateException("The primary column must be an ENTITY_ID or STRING column");
 		}
+
 		for(int i=0;i<input.getWidth();i++) {
 			ColumnDescription columnDescription = input.getColumnDescription(i);
 			columns[i] = new PPColumn(columnDescription.getName());
 			columns[i].setParser(columnDescription.getType().createParser());
 		}
 	}
-	
+
+	public void write(HCFile outFile) throws IOException {
```

Comment:
Die Methoden Signatur ist hier unsauber, weil man der Methode auch ein HCFile im READ mode übergeben kann.

Reply:
Ich hab hier einen Test eingebaut ob es in read geöffnet ist. Die alte variante war mMn noch blöder aber ich sehe hier keinen trivialen Weg das zu garantieren.  Ich hab auch ein todo in die HCFile klasse geschrieben aber so wie sie aktuell ist gefällt sie mir eigentlich nicht